test/e2e/kubectl/kubectl.go

```diff
@@ -163,6 +164,24 @@ properties:
               pattern: in-tree|out-of-tree
               type: string`)
 
+var schemaFooEmbedded = []byte(`description: Foo CRD with an embedded resource
+type: object
+properties:
+  spec:
+    type: object
+    properties:
+      template:
+        type: object
+        x-kubernetes-embedded-resource: true
+        properties:
```

Comment:
I can't remember if explicitly defining a `metadata` property with a `name` subproperty made a difference or not in the parent issue... might consider doing that here, then exercising an implicit field (`namespace`), an explicit field (`name`), and an unknown field (`unknownField`) when writing the custom resource

Reply:
Discussed offline, I now have the CRD explicitly defining the `name` subproperty within the `metadata `property for both root and embedded objects. I’m also passing both `name` and `namespace` values for root and embedded `metadata` fields on all requests to confirm that these are handled properly.

For now, unknown fields are passed for root and embedded metadata objects separately until I can find a better way to validate multiple error messages that is consistent both now (with the client performing the unknown field validation) and after unknown field validation is performed default on the server.